### PR TITLE
[SYCL][E2E] Add sycl-toolchain as a dependency

### PR DESCRIPTION
**Pull request by victor-eds**
`check-sycl-e2e-*` test the SYCL toolchain, so we should make sure it is up-to-date.

### CODE REVIEW

File: sycl/test-e2e/CMakeLists.txt [hunk 1]
```diff
@@ -36,6 +36,10 @@ find_package(Python3 REQUIRED COMPONENTS Interpreter)
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/lit.site.cfg.py.in"
                "${CMAKE_CURRENT_BINARY_DIR}/lit.site.cfg")
 
+list(APPEND SYCL_E2E_TEST_DEPS
+  sycl-toolchain
+  )
+
```
Comment (aelovikov-intel) [inline sycl/test-e2e/CMakeLists.txt:42]: I think we to guard that to skip it in standalone mode (see line 7 for its definition).

Comment (victor-eds) [reply sycl/test-e2e/CMakeLists.txt]: Good point. I added the guard.

File: sycl/test-e2e/CMakeLists.txt [hunk 2]
```diff
@@ -58,6 +64,7 @@ if(SYCL_TEST_E2E_TARGETS)
     add_custom_target(${TARGET}
       COMMAND ${Python3_EXECUTABLE} ${LLVM_LIT} ${SYCL_E2E_TESTS_LIT_FLAGS} --param sycl_be=${TARGET_BE} --param target_devices=${TARGET_DEVICES} .
       COMMENT "Running the SYCL tests for ${TARGET} backend"
+      DEPENDS ${SYCL_E2E_TEST_DEPS}
```
Comment (aelovikov-intel) [inline sycl/test-e2e/CMakeLists.txt:67]: I'm not sure how CMake reacts to an "uninitialized variable", let's see what CI will tell us.

Comment (victor-eds) [reply sycl/test-e2e/CMakeLists.txt:67]: According [to the documentation](https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#variable-references):

> A variable reference has the form ${<variable>} and is evaluated inside a [Quoted Argument](https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#quoted-argument) or an [Unquoted Argument](https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#unquoted-argument). A variable reference is replaced by the value of the specified variable or cache entry, or if neither is set, by the empty string.

**Review by aelovikov-intel — APPROVED** [review]

File: sycl/test-e2e/CMakeLists.txt (hunk 2 above)
Comment (aelovikov-intel) [reply sycl/test-e2e/CMakeLists.txt:67]: Thanks! I also see some testing has successfully finished, so the patch looks good.

Thanks for taking care of this!

PS, there might be one other issue with this non-standalone configuration where we inherit the compilation flags from the project, but I didn't have cycles to look into that, so JFYI.

File: sycl/test-e2e/CMakeLists.txt [hunk 3]
```diff
@@ -36,6 +36,12 @@ find_package(Python3 REQUIRED COMPONENTS Interpreter)
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/lit.site.cfg.py.in"
                "${CMAKE_CURRENT_BINARY_DIR}/lit.site.cfg")
 
+if(NOT SYCL_TEST_E2E_STANDALONE)
+  list(APPEND SYCL_E2E_TEST_DEPS
+    sycl-toolchain
```
Comment (aelovikov-intel) [inline sycl/test-e2e/CMakeLists.txt:41]: Strictly speaking, there are other test utilities deps like `FileCheck`, but should normally be an issue on the very first invocation only, and we can address it later.

Comment (victor-eds) [reply sycl/test-e2e/CMakeLists.txt:41]: I can add `FileCheck` and maybe also `not` if you want, but, as you say, we can address that in a different one.

Comment (aelovikov-intel) [reply sycl/test-e2e/CMakeLists.txt:41]: No, I think we're good to call @intel/llvm-gatekeepers to merge this one.

**Pull request merged**